src/Common/TagStoreInterface.php

```diff
@@ -4,15 +4,66 @@
 
 interface TagStoreInterface
 {
-    public function tagObject($objectId, array $tags);
+    /**
+     * Associate the given tags to cache item identified by $itemId.
+     *
+     * Example: associate a HTTP cache identifier (e.g. a content-digest in the
+     *          case of the Symfony HTTP cache) with a set of tags.
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function tagItemId($itemId, array $tags);
 
-    public function untagObject($objectId, array $tags);
+    /**
+     * Unassociate the given tags from the cache item identified by $itemId.
+     *
+     * NOTE: This is difficult for an so-called "cache based" implementation, is it necessary?
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function untagItemId($itemId, array $tags);
```

Comment:
Given that what makes `untagItemId` and `getTagsForItemIds` difficult to implement also applies to the needed `removeItemIds`, I'm leaning more towards keeping them. This is a generic tag store, theoretically it could be used e.g. for storing tags for blog entries, or git branches, or whatever, in which case those methods become a bit more useful.

Had it been the case that we could have skipped `removeItemIds`, I might have suggested we split it into two interfaces instead, but that doesn't seem practical currently.